torch/utils/data/dataloader.py

```diff
@@ -234,11 +262,23 @@ def _shutdown_workers(self):
         if not self.shutdown:
             self.shutdown = True
             self.done_event.set()
+            # if process_thread is waiting to put
+            if not self.data_queue.empty():
```

Comment:
SimpleQueue doesn't have an `empty()` method

Reply:
I'm pretty sure that it does... https://docs.python.org/2/library/multiprocessing.html#multiprocessing.multiprocessing.queues.SimpleQueue.empty